.github/workflows/lint-and-test.yml

```
@@ -0,0 +1,45 @@
+name: lint and test
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+env:
+  TZ: Asia/Tokyo
+
+jobs:
+  lint:
+    name: lint
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out
+        uses: actions/checkout@v3
+      - name: Set up Go
+        uses: actions/setup-go@v3
+        with:
+          go-version: 1.22
+          cache: true
+      - name: golangci-lint
+        uses: golangci/golangci-lint-action@v3
```

[nits] 最新が6だったので
https://github.com/golangci/golangci-lint-action
```suggestion
        uses: golangci/golangci-lint-action@v6
```